### PR TITLE
Elasticsearch: Fix script fields in query editor

### LINKED ISSUE #23766: Elasticsearch: Scripted field deprecated key names
**What happened**:

In version 5.6.0 of ElasticSearch there was a change of scripted fields keys, "inline" was changed to "source", and "stored" to "id". Grafana is still using the "inline" script when using a scripted field, which triggers a Deprecated warning in ElasticSearch. The support for old key names will eventually get removed, which will break Grafana graphs which use scripted fields.

Example:
Setting script value to:
`_value / 60
`Results in a query like:
`body:"{"search_type":"query_then_fetch","ignore_unavailable":true,"index":["flows-*"]} {"size":0,"query":{"bool":{"filter":[{"range":{"@timestamp":{"gte":1587525071624,"lte":1587535871624,"format":"epoch_millis"}}},{"query_string":{"analyze_wildcard":true,"query":"netflow.ipv4_src_addr: x.x.x.x"}}]}},"aggs":{"3":{"terms":{"field":"netflow.ipv4_src_addr","size":10,"order":{"_key":"desc"},"min_doc_count":1},"aggs":{"2":{"date_histogram":{"interval":"1m","field":"@timestamp","min_doc_count":0,"extended_bounds":{"min":1587525071624,"max":1587535871624},"format":"epoch_millis"},"aggs":{"1":{"sum":{"field":"netflow.in_bits","missing":0,"script":{"inline":"_value / 60"}}}}}}}}} "`

The graph works, but ElasticSearch logs a warning:
`[2020-04-20T12:47:42,994][WARN ][o.e.d.c.ParseField       ] [node0] Deprecated field [inline] used, expected [source] instead
`

Grafana has to use "inline" for ElasticSearch prior to 5.6.0 and "source" for ElasticSearch version 5.6.0+.

Reference: https://www.elastic.co/guide/en/elasticsearch/reference/5.6/release-notes-5.6.0.html
Related GitHub change: https://github.com/elastic/elasticsearch/pull/25127

**What you expected to happen**:

No deprecated warnings.

**How to reproduce it (as minimally and precisely as possible)**:
- create a graph in Grafana from ElasticSearch
- select metric: sum
- click options
- put in a script, like: `_value / 1`
- open the graph
- watch elasticsearch logs

**Anything else we need to know?**:

**Environment**:
- Grafana version: 6.7.1
- Data source type & version: ElasticSearch 7.1
- OS Grafana is installed on: Debian Linux 10
- User OS & Browser: OSX 10.15.3
- Grafana plugins: N/A
- Others:

### LINKED ISSUE #31531: Upgrading from 7.3.x to 7.4.x has issue with elasticsearch inline scripts in queries
**What happened**: After upgrading to Grafana 7.4.2, existing elasticsearch metrics queries 'lose' their inline scripts.  The panels still work, but when you edit them, you see `Object [Object]` in the script field (and the queries then break).  Exporting the json of each panel allows me to restore the scripts manually, but it's not possible to do this through the UI.

**What you expected to happen**: Existing queries with inline scripts remain editable after the upgrade.

**How to reproduce it (as minimally and precisely as possible)**: Create an elasticsearch panel query with an inline script (e.g. `_value * 8` to convert from bytes to bits) using Grafana < 7.4 (I last edited this dashboard with 7.3.x).  Save it. Upgrade the grafana instance to 7.4.2.  Edit the query - you should see that your script is replaced by `[object Object]`, and upon saving the panel the query breaks.

**Anything else we need to know?**:

I compared the Panel JSON and the old panel has this:
```
      "metrics": [
        {
          "$$hashKey": "object:121",
          "field": "system.network.out.bytes",
          "hide": true,
          "id": "1",
          "inlineScript": "_value * 8",
          "meta": {},
          "settings": {
            "script": {
              "inline": "_value * 8"
            }
          },
          "type": "max"
        },
        {
          "$$hashKey": "object:228",
          "field": "1",
          "id": "3",
          "meta": {},
          "pipelineAgg": "1",
          "settings": {
            "unit": "1s"
          },
          "type": "derivative"
        }
      ],
      "query": "host.name:\"$hostname\" AND metricset.name:network",
      "refId": "B",
      "timeField": "@timestamp"
```

The new panel after re-entering the script and saving looks like this:

```
      "metrics": [
        {
          "$$hashKey": "object:121",
          "field": "system.network.out.bytes",
          "hide": true,
          "id": "1",
          "inlineScript": "_value * 8",
          "meta": {},
          "settings": {
            "script": "_value * 8"
          },
          "type": "max"
        },
        {
          "$$hashKey": "object:228",
          "field": "1",
          "id": "3",
          "meta": {},
          "pipelineAgg": "1",
          "settings": {
            "unit": "1s"
          },
          "type": "derivative"
        }
      ],
      "query": "host.name:\"$hostname\" AND metricset.name:network",
      "refId": "B",
      "timeField": "@timestamp"
```

Note the change from `"script": { "inline": "<script>" }` to `"script" : "<script>"`

**Environment**:
- Grafana version: 7.4.2
- Data source type & version: Elasticsearch (built-in)
- OS Grafana is installed on: RHEL7
- User OS & Browser: Win10, Edge Chromium
- Grafana plugins: n/a
- Others:

### PR DESCRIPTION
**Pull request by Elfo404**
**What this PR does / why we need it**:
before `7.4`, `script` settings were stored like
```
settings: {
  script: {
    inline: "value"
  }
}
```

during the react refactoring this was accidentally changed to
```
settings: {
  script: "value"
}
```
which for Grafana use cases is totally equivalent (both syntaxes were/are supported by Elasticsearch)

this caused an issue with the new query editor, where the content of the settings field for the script would appear as `[object Object]`.

This PR doesn't really fix the way we store this setting (eg, rolling back to the previous version) but rather consolidate the new syntax as it fully functional and also fixes deprecation warnings as reported in  #23766

**Which issue(s) this PR fixes**:

Fixes #31531

The bug (not this PR) fixed by this PR incidentally fixed #23766

**Special notes for your reviewer**:

to properly test this works, you can use the following query model which has metric aggregations with both syntaxes.
Without modifying the query, all fields should get retrieved in every ES version (2,5,6,7) and all the script settings should be displayed correctly:

Screenshot 2021-03-04 at 14 19 47

```json
  "targets": [
    {
      "alias": "",
      "bucketAggs": [
        {
          "field": "@timestamp",
          "id": "2",
          "settings": {
            "interval": "auto",
            "min_doc_count": 0,
            "trimEdges": 0
          },
          "type": "date_histogram"
        }
      ],
      "metrics": [
        {
          "field": "select field",
          "id": "1",
          "type": "count"
        },
        {
          "id": "3",
          "settings": {
            "script": {
              "inline": "100"
            }
          },
          "type": "avg"
        },
        {
          "id": "4",
          "settings": {
            "script": {
              "inline": "100"
            }
          },
          "type": "sum"
        },
        {
          "id": "5",
          "settings": {
            "script": {
              "inline": "100"
            }
          },
          "type": "max"
        },
        {
          "id": "6",
          "settings": {
            "script": {
              "inline": "100"
            }
          },
          "type": "min"
        },
        {
          "id": "7",
          "meta": {
            "std_deviation_bounds_lower": true,
            "std_deviation_bounds_upper": true
          },
          "settings": {
            "script": {
              "inline": "100"
            }
          },
          "type": "extended_stats"
        },
        {
          "id": "8",
          "settings": {
            "percents": ["25", "50", "75", "95", "99"],
            "script": {
              "inline": "100"
            }
          },
          "type": "percentiles"
        },
        {
          "id": "9",
          "pipelineVariables": [
            {
              "name": "var1",
              "pipelineAgg": ""
            }
          ],
          "settings": {
            "script": {
              "inline": "100"
            }
          },
          "type": "bucket_script"
        },
        {
          "id": "10",
          "settings": {
            "script": "100"
          },
          "type": "avg"
        },
        {
          "id": "11",
          "settings": {
            "script": "100"
          },
          "type": "sum"
        },
        {
          "id": "12",
          "settings": {
            "script": "100"
          },
          "type": "max"
        },
        {
          "id": "13",
          "settings": {
            "script": "100"
          },
          "type": "min"
        },
        {
          "id": "14",
          "meta": {
            "std_deviation_bounds_lower": true,
            "std_deviation_bounds_upper": true
          },
          "settings": {
            "script": "100"
          },
          "type": "extended_stats"
        },
        {
          "id": "15",
          "settings": {
            "percents": ["25", "50", "75", "95", "99"],
            "script": "100"
          },
          "type": "percentiles"
        },
        {
          "id": "16",
          "pipelineVariables": [
            {
              "name": "var1",
              "pipelineAgg": ""
            }
          ],
          "settings": {
            "script": "100"
          },
          "type": "bucket_script"
        }
      ],
      "query": "",
      "refId": "A",
      "timeField": "@timestamp"
    }
  ]
```

### CODE REVIEW

**Comment by Elfo404** [comment]
Converted to draft as there's also the backend involved that needs to be checked. sorry for the spam.

**Comment by Elfo404** [comment]
apparently, the problem with alerting is a different one which is not strictly related to this PR: https://github.com/grafana/grafana/issues/31694

File: public/app/plugins/datasource/elasticsearch/components/MetricPicker.tsx [hunk 1]
```diff
@@ -23,12 +23,16 @@ interface Props {
...
-  <Segment
-    className={cx(className, noWrap)}
-    options={toOptions(options)}
-    onChange={onChange}
-    placeholder="Select Metric"
-    value={!!value ? toOption(options.find((option) => option.id === value)!) : null}
-  />
-);
+export const MetricPicker: FunctionComponent<Props> = ({ options, onChange, className, value }) => {
+  const selectedOption = options.find((option) => option.id === value);
```
Comment (Elfo404) [inline public/app/plugins/datasource/elasticsearch/components/MetricPicker.tsx:27]: Unrelated change, but this was breaking in certain situations when expanding the options for metrics with no fields

**Review by ivanahuckova — APPROVED** [review]
LGTM! Thanks for adding the query model to test this. It really speed up the testing. Super appreciated. 🙏 Regarding PR, everything works as expected. You have added very helpful comments. Code is very readable. Good job! 🚀

**Pull request merged**